mapbox/vs30/scripts/basin_z_values/gen_sites_in_basin.py

```diff
@@ -0,0 +1,160 @@
+        dy2 = point[1] - polygon[jj][1]
+
+        F = (dx - dx2) * dy - dx * (dy - dy2)
+        if 0.0 == F and dx * dx2 <= 0 and dy * dy2 <= 0:
+            return 2
+
+        if (dy >= 0 and dy2 < 0) or (dy2 >= 0 and dy < 0):
+            if F > 0:
+                intersections += 1
+            elif F < 0:
+                intersections -= 1
+
+        jj += 1
+
+    return intersections != 0
+
+
+@njit(parallel=True)
+def is_inside_postgis_parallel(points, polygon):
+    ln = len(points)
+    D = np.empty(ln, dtype=numba.boolean)
+    for i in numba.prange(ln):
+        D[i] = is_inside_postgis(polygon, points[i])
+    return D
+
+
+def get_args():
+
+    parser = argparse.ArgumentParser()
+    arg = parser.add_argument
+    arg(
+        "--outfile",
+        help="Path to the output CSV file",
+        default=Path.cwd() / "sites_in_basin.csv",
+    )
+    arg(
+        "--remove_not_in_basin",
+        help="Remove stations not in basin",
+        action="store_true",
+    )
+    arg("--version", help=f"NZVM version. Default is {BASIN_VER}", default=BASIN_VER)
+    parser = formats.station_file_argparser(parser=parser)
+    return parser.parse_args()
+
+
+if __name__ == "__main__":
+    args = get_args()
+    try:
+        basin_outlines = basin_outlines_dict[arg.version]
+    except KeyError:
+        raise ValueError(f"Basin outlines for version {args.version} not found")
+
+    # Load the boundaries
+    boundaries = {cur_ffp.stem: np.loadtxt(cur_ffp) for cur_ffp in basin_outlines}
+
+    if args.stat_file is None:
+        # remove points in ocean
+        maskvalues = [0, 1, 1, 1, 1]  # ocean / land / lake / island / pond.
+        land_mask = pygmt.grdlandmask(
+            region="NZ",
+            spacing=f"{GRID_SPACING}e/{GRID_SPACING}e",
+            maskvalues=maskvalues,
+            resolution="f",
+        )
+        land_mask_array = land_mask.values  # M x N boolean values
+
+        # Create an index array where land_mask_array is equal to 1.0 (land)
+        land_indices = np.argwhere(
+            land_mask_array == 1.0
+        )  # returns (lat_index, lon_index) pairs
+        # Use the land_indices to extract longitude and latitude values for land areas
+        land_lon = land_mask.lon.values[land_indices[:, 1]]
+        land_lat = land_mask.lat.values[land_indices[:, 0]]
+
+        ll_points = np.column_stack(
+            (land_lon, land_lat)
+        )  # zip them into an array of (lon, lat) pairs
+        stat_names = None
+    else:
+        station_df = formats.load_generic_station_file(
+            args.stat_file,
+            args.stat_name_col,
+            args.lon_col,
+            args.lat_col,
+            args.other_cols,
+            args.other_names,
+            args.sep,
+            args.skiprows,
+        )
+        ll_points = station_df[["lon", "lat"]].values
+        stat_names = station_df.index.values
+
+    nztm_points = geo.wgs_nztm2000x(ll_points)  # convert to NZGD lon and NZGD lat
```

Comment:
more an informational comment but once https://github.com/ucgmsim/qcore/pull/303 is merged, the new `qcore.coordinates.wgs_depth_to_nztm` function is a better choice because it uses pyproj for the coordinates conversion instead of our random code.